site/src/modules/dashboard/Navbar/Navbar.tsx

```diff
@@ -25,12 +25,19 @@ export const Navbar: FC = () => {
 	const canViewConnectionLog =
 		featureVisibility.connection_log && permissions.viewAnyConnectionLog;
 
+	const uniqueLinks = appearance.support_links
+		? appearance.support_links.filter(
+				(link, index, self) =>
+					index === self.findIndex((l) => l.name === link.name),
+			)
+		: undefined;
```

Comment:
I'm wondering if it might be better to make the `supportLinks` prop required for `NavbarView`, and then change the checks to see whether the array is empty. That way we don't have to deal with both the empty and undefined cases

And if we do that, we can simplify `uniqueLinks` to:
```tsx
const uniqueLinks = [...new Set(appearance.support_links)]
```

Reply:
To be honest, I don't know if I've _ever_ seen the third `self` argument used in professional code before

Reply:
specifically, here's my suggestion:

```suggestion
	const uniqueLinks = new Map<string, LinkConfig>();
	for (const link of appearance.support_links) {
		if (!uniqueLinks.has(link.name)) {
			uniqueLinks.set(link.name, link);
		}
	}
	// and later get the list by doing `[...uniqueLinks.values()]`
```

tho I still think that data hygiene concerns like this belong more in the backend than that frontend

it's also worth noting that this "call a function if it exists, and give me undefined if not" pattern is common enough that it has it's own operator: `?.`

Reply:
Oh right, duh. I forgot that these were objects. Another option would be to do this:
```ts
const uniqueLinks = [
  ...new Map(
    appearance.supportLinks?.map((l) => [l.name, l])
  ).values()
];
```
Though the nuances change a little bit. Kayla's suggestion always keeps the first entry that is encountered when you have a conflict, while mine keeps whichever one occurs later in the array

Reply:
I wouldn't touch the backend side now, but we can improve it in a follow-up. I updated the frontend side as you suggested. Please let me know your thoughts.